dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -584,9 +585,10 @@ public ResourceSupport findRel(HttpServletRequest request, HttpServletResponse r
         RestAddressableModel modelObject = null;
         try {
             modelObject = repository.upload(request, apiCategory, model, id, uploadfile);
-        } catch (Exception e) {
-            log.error(e.getMessage(), e);
-            return ControllerUtils.toEmptyResponse(HttpStatus.INTERNAL_SERVER_ERROR);
+        } catch (SQLException | IOException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
Honestly, I'm not 100% sure here. We usually re-throw exception in this way and you will see that broadly on the source code. According to the javadoc the single constructor will call this one with a little different "message", instead than `getMessage` the `toString` of the cause will be called.

Reply:
@abollini : The reason I don't like `e.getMessage()` alone is twofold: (1) it can be null, and (2) it may not always be as informative as `e.toString()` (toString contains both the message and the classname).   As noted in the [JavaDocs of `RuntimeException(Throwable cause)`](https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html): 

> Constructs a new runtime exception with the specified cause and a detail message of (cause==null ? null : cause.toString()) (which typically contains the class and detail message of cause). This constructor is useful for runtime exceptions that are little more than wrappers for other throwables.

So, if all you want to do is wrap an exception in a RuntimeException, you really should be using `throw new RuntimeException(e);`  (you are correct, we have a lot of code that doesn't do this right now, but should be updated someday).  However, if you want to provide more contextual information around the error, this could be updated to say something like:
`throw new RuntimeException("Error " + e.getMessage() + uploading file to " + model + " with ID= " +id, e);`

Reply:
@tdonohue oh yes it makes sense. Thanks to take the time to explain the motivation we will try to stick to this rule at least for all the new code and when touched update existing code as wel